libvmi/driver/xen/xen_events.c

```diff
@@ -1878,8 +1990,18 @@ static
 status_t wait_for_event_or_timeout(vmi_instance_t vmi, unsigned long ms, bool *needs_unmasking)
 {
     xen_events_t *xe = xen_get_events(vmi);
+#ifdef HAVE_LIBXENSTORE
+    int rc = 0;
+
+    if ( vmi->init_flags & VMI_INIT_DOMAINWATCH )
```

Comment:
I would prefer if we didn't have this bitmask in here, but rather `xe` had a field that recorded the size of the `fd` array, and we passed that in to `poll`. This is somewhat in the fast-path during events so adding an extra bitmask and if to every call to this function is going to add unnecessary overhead.

Reply:
Is a new field mandatory? I think we can use a macro for COUNT_OF or something like 
switch ( poll(xe->fd, sizeof(xe->fd)/sizeof(xe->fd[0]), ms) )

Reply:
Well, afaict the array is always of size two but you only use the first array index with events without domain watcher. So for what you propose you would have to grow the array size dinamically.

Reply:
I've added the fd_size field in the xe structure